lib/config.js

```diff
@@ -14,24 +21,93 @@ function getNodeVersion () {
   return [Number(version[1]), Number(version[2])];
 }
 
-async function getGitRev () {
-  let rev = null;
+async function updateBuildInfo () {
+  let sha;
+  try {
+    sha = await getGitRev();
+    if (_.isEmpty(sha)) {
+      logger.warn(`Cannot retrieve git revision for Appium version ${APPIUM_VER}`);
+      return;
+    }
+  } catch (err) {
+    logger.warn(`Cannot retrieve git revision for Appium version ${APPIUM_VER}. ` +
+      `Original error: ${err.message}`);
+    return;
+  }
+
+  BUILD_INFO['git-sha'] = sha;
   try {
-    let {stdout} = await exec("git", ["rev-parse", "HEAD"], {rootDir});
-    rev = stdout.trim();
-  } catch (ign) {}
-  return rev;
+    const built = await getGitTimestamp(sha);
```

Comment:
this is still going to block on a network request, isn't it?

we probably also want to `catch` these requests and ignore failures so appium can run without internet

Reply:
it should not. These methods are called from `updateBuildInfo`, which is not awaited